chat_test.go

```diff
@@ -764,3 +764,284 @@ func TestFinishReason(t *testing.T) {
+    "created": 1731313962,
+    "model": "gpt-4o-audio-preview-2024-10-01",
+    "system_fingerprint": "fp_49254d0e9b",
+    "choices": [
+        {
+            "index": 0,
+            "delta": {
+                "audio": {
+                    "transcript": ","
+                }
+            },
+            "finish_reason": null
+        }
+    ]
+}`,
+		},
+		{
+			response: openai.ChatCompletionStreamResponse{
+				ID:                "chatcmpl-ASK8gd4isaVFw7qClLmtrcwWvka7p",
+				Object:            "chat.completion.chunk",
+				Created:           1731313962,
+				Model:             openai.GPT4oAudioPreview20241001,
+				SystemFingerprint: "fp_49254d0e9b",
+				Choices: []openai.ChatCompletionStreamChoice{
+					{
+						Index: 0,
+						Delta: openai.ChatCompletionStreamChoiceDelta{
+							Role: openai.ChatMessageRoleAssistant,
+							Audio: &openai.ChatCompletionStreamChoiceDeltaAudio{
+								ID:   "audio_6731c12b1c148190bb8db8af1330221a",
+								Data: "base64 encoded data",
+							},
+						},
+					},
+				},
+			},
+			want: `{
+    "id": "chatcmpl-ASK8gd4isaVFw7qClLmtrcwWvka7p",
+    "object": "chat.completion.chunk",
+    "created": 1731313962,
+    "model": "gpt-4o-audio-preview-2024-10-01",
+    "system_fingerprint": "fp_49254d0e9b",
+    "choices": [
+        {
+            "index": 0,
+            "delta": {
+                "role": "assistant",
+                "content": null,
+                "refusal": null,
+                "audio": {
+                    "id": "audio_6731c12b1c148190bb8db8af1330221a",
+                    "data": "base64 encoded data"
+                }
+            },
+            "finish_reason": null
+        }
+    ]
+}`,
+		},
+		{
+			response: openai.ChatCompletionStreamResponse{
+				ID:                "chatcmpl-ASK8gd4isaVFw7qClLmtrcwWvka7p",
+				Object:            "chat.completion.chunk",
+				Created:           1731313962,
+				Model:             openai.GPT4oAudioPreview20241001,
+				SystemFingerprint: "fp_49254d0e9b",
+				Choices: []openai.ChatCompletionStreamChoice{
+					{
+						Index: 0,
+						Delta: openai.ChatCompletionStreamChoiceDelta{
+							Audio: &openai.ChatCompletionStreamChoiceDeltaAudio{
+								Data: "base64 encoded data",
+							},
+						},
+					},
+				},
+			},
+			want: `{
+    "id": "chatcmpl-ASK8gd4isaVFw7qClLmtrcwWvka7p",
+    "object": "chat.completion.chunk",
+    "created": 1731313962,
+    "model": "gpt-4o-audio-preview-2024-10-01",
+    "system_fingerprint": "fp_49254d0e9b",
+    "choices": [
+        {
+            "index": 0,
+            "delta": {
+                "audio": {
+                    "data": "base64 encoded data"
+                }
+            },
+            "finish_reason": null
+        }
+    ]
+}`,
+		},
+	}
+
+	for _, c := range cases {
+		var expected openai.ChatCompletionStreamResponse
+		err := json.Unmarshal([]byte(c.want), &expected)
+		checks.NoError(t, err)
+		checks.Equal(t, c.response, expected)
+	}
+}
```

Comment:
Could you please add an integration test? It's pretty hard to review new things these days!